src/angular/form-elements/dropdown/dropdown.component.ts

```diff
@@ -1,28 +1,200 @@
+
+    /**
+     * Export DropDownOptionType enum so we can use it on the template
+     */
+    public cIDropDownOptionType = DropDownOptionType;
+
+    /**
+     * Configure unselectable option types
+     */
+    private unselectableOptions = [DropDownOptionType.Disable, DropDownOptionType.Header, DropDownOptionType.HorizontalLine];
+
+    public isGroupDesign = false;
+
+    constructor(private view:ElementRef){}
+
+    ngOnInit(): void {
+        if(this.options){
+            if(this.options.find(option => option.type === DropDownOptionType.Header)){
+                this.isGroupDesign = true;
+            }
+        }
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if(changes.value){
+            this.setValueIfExist(this.value);
+        }
+    }
+
+    public isValid(){
+        return !this.error;
+    }
+
+    /**
+     * Validate when required is enabled
+     */
+    public validateDropDown(): void{
+        if(!this.disabled && this.required && (!this.value || this.value === '')){
+            this.error = true;
+            return;
+        }
+        this.error = false;
+    }
+
+    /**
+     * Set option as selected and saves it's index on the list
+     * @param index - number
+     * @param option - IDropDownItem or string
+     */
+    public selectOption(index: number, option:any):void{
+        const unselectable = this.unselectableOptions.find(optionType => option.type === optionType);
+        if(unselectable){
+            return;
+        }
+        this.selectedIndex = index;
+        this.updateSelected(option.label || option, option.value || option);
+    }
+
+    /**
+     * Update the value and label of the drop down with new ones
+     * @param label
+     * @param value
+     */
+    private updateSelected(label: string, value: string, index?: number){
+        this.selectedLabel = label;
+        this.value = value;
+        this.baseEmitter.next(this.value);
+        this.selectedIndex = index || this.selectedIndex;
+        this.validateDropDown();
+    }
+
+    /**
+     * Get the label of the selected option
+     */
+    public getSelectedLabel(): string{
+       return this.selectedLabel || null;
+    }
+
+    public isBottomVisible(){
+        const windowPos = window.innerHeight + window.pageYOffset;
+        const dropDownPos = this.dropDownWrapper.nativeElement.offsetTop + this.dropDownWrapper.nativeElement.offsetHeight + 244;
+        console.log("dropDownWrapper", this.dropDownWrapper.nativeElement.offsetHeight);
+        return windowPos > dropDownPos;
+    }
+
+    /**
+     * Toggle show/hide drop-down list
+     */
+    public toggleDropdown(){
+        if(!this.disabled){
+            this.show = !this.show;
+        }
+    }
+
+    /**
+     * When users clicks outside the drop-down it will be closed
+     */
+    public onClickOutside(event){
+        if(!event.target.classList.contains('js-sdc-dropdown--toggle-hook')){
+            this.show = false;
+        }
```

Comment:
Try click on an unselectable option - my guess that the dropdown list will be closed.

Better approach for this:
Put an element reference on the `<ul>` tag contains the options list (with class `'sdc-dropdown__options-wrapper'`).
```
<ul #optionsContainerElem class="sdc-dropdown__options-wrapper" ...>
```
```
public onClickOutside(event){
    if (!this.optionsContainerElement.contains(event.target)) {
        this.show = false;
    }
}
```
This will prevent closing the list when clicking inside the list. Therefore, you should also close the list explicitly when selecting a valid option.